src/screens/authentication/UsernameRequest/index.js

```diff
@@ -1,22 +1,61 @@
 import React, { useState } from 'react'
-import { KeyboardAvoidingView, Image, TextInput, ActivityIndicator } from 'react-native'
-import { Text, Button } from '~/components'
-import Background from '~/helpers/Background'
+import { KeyboardAvoidingView, TextInput, ActivityIndicator, View } from 'react-native'
+import { SvgXml } from 'react-native-svg'
+import { Body, Description, Subtitle } from '~/components/texts'
+import { PrimaryButton, GrayButton } from '~/components/buttons'
 import * as AuthenticationService from '~/authenticationService'
 import * as DatabaseService from '~/databaseService'
 import NavigationService from '~/navigationService'
-import DrCannabis from '~/assets/images/DrCannabis.png'
-import styles from './styles'
+import usernameIcon from '~/assets/images/UsernameRequest/usernameIcon.svg'
+import styles, { PLACEHOLDER_TEXT_COLOR, ICON_WIDTH, ICON_HEIGHT } from './styles'
 
 const SubmitIndicator = ({ submitting }) => (
   submitting &&
     <ActivityIndicator
       style={styles.submitIndicator}
-      size='large' />
+      size='large'
+    />
 )
 
 const Error = ({ error }) => (
-  error && <Text fontVariant='h3' style={styles.error}>{error}</Text>
+  error && (
+    <Body
+      secondary
+      style={styles.error}
+    >
+      {error}
+    </Body>
+  )
+)
+
+const SubmitButton = ({ value, onPress }) => {
+  if (value === '') {
+    return (
+      <GrayButton>
+        <Description white>Continuar</Description>
+      </GrayButton>
+    )
+  } else {
+    return (
+      <PrimaryButton
+        onPress={onPress}
+      >
+        <Description white>Continuar</Description>
+      </PrimaryButton>
+    )
+  }
+}
```

Comment:
The `value` prop is only used for the `if` statement?

Reply:
Yes

Reply:
:thinking: I think that prop should be a `boolean` with a more descriptive name, like `disabled`.